modules/coreos-layering-on-cluster-opt-in.adoc

```diff
@@ -0,0 +1,136 @@
+// Modules included in the following assemblies:
+//
+// * post-installation_configuration/coreos-layering.adoc
+
+:_content-type: PROCEDURE
+[id="coreos-layering-on-cluster-opt-in_{context}"]
+= Opting in to create on-cluster {op-system} builds
+
+In order to create on-cluster builds, you need to opt in. This process involves adding a specific annotation to any specified `MachineConfigPool` objects to signal that the Machine Config Operator (MCO) should begin an on-cluster build.
+
+.Prerequisites
+
+. You have a running {product-title} cluster.
+. You installed the OpenShift CLI (`oc`).
+. You are logged in to the cluster as a user with administrative privileges.
+. You have at least one SSH key pair configured either using the `install-config.yaml` file or a machine config file.
+. You have access to an image registry and those permissions have been added to the global pull secret.
+
+.Procedure
+
+. Perform the initial setup by obtaining the following information.
+
+.. Get the name of the pull secret for the base {op-system} image and the extensions container and add it as a secret in the MCO namespace. You can do this by cloning the global pull secret:
++
+[source,terminal]
+----
+$ oc create secret docker-registry global-pull-secret-copy && \
+  -n openshift-machine-config-operator && \
+  --from-file=.dockerconfigjson=<(oc get secret/pull-secret -n openshift-config -o json | jq -r '.data[".dockerconfigjson"] | @base64d')
+----
+
+.. Get the final image pullspec from your registry. For example:
++
+[source,terminal]
+----
+$ podman pull quay.io/my-registry/custom-image:latest
+----
++
+Any tags you provide are ignored. Instead, images are tagged with the name of the rendered machine config that was used to produce the image. For example:
++
+[source,text]
+----
+quay.io/my-registry/custom-image:rendered-worker-dd37180cefa6a1e88834966330c0c028
+----
+
+.. Get the name of the push secret to enable the final {op-system} image that will be pushed using the above pullspec. This must be created as a secret within the MCO namespace. For example:
++
+[source,terminal]
+----
+$ <command to copy final image push secret to MCO namespace>
+----
+
+.. Optional: Make a note of the original {op-system} image. You will need this to revert the node, in the case that you want to opt-out:
++
+[source,terminal]
+----
+$ oc get configmap/machine-config-osimageurl && \
```

Comment:
This will not work, please drop the `&&`, similarly below. Is someone testing these intructions?